[CI] TransformUsageIT.testUsage fails with incorrect stats call count

In https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob+fast+part2/3953/consoleFull / https://gradle-enterprise.elastic.co/s/nobrqofhrecye/tests/vkcr26w2ldob6-j74o4svwlg7nm

The `TransformUsageIT.testUsage` test failed with:

```
java.lang.AssertionError: 
Incorrect stat trigger_count expected:<3.0> but was:<2.0>
at __randomizedtesting.SeedInfo.seed([5671C4D5CB7154:BCF1ED1069B7A9DD]:0)
at org.junit.Assert.fail(Assert.java:88)
at org.junit.Assert.failNotEquals(Assert.java:834)
at org.junit.Assert.assertEquals(Assert.java:553)
at org.elasticsearch.xpack.transform.integration.TransformUsageIT.lambda$testUsage$2(TransformUsageIT.java:105)
at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:881)
...
```

I was unable to reproduce this on master with:

```
./gradlew ':x-pack:plugin:transform:qa:single-node-tests:integTestRunner' --tests "org.elasticsearch.xpack.transform.integration.TransformUsageIT.testUsage" -Dtests.seed=5671C4D5CB7154 -Dtests.security.manager=true -Dtests.locale=th-TH -Dtests.timezone=Africa/Bissau -Dcompiler.java=13
```

--- a/x-pack/plugin/transform/qa/single-node-tests/src/test/java/org/elasticsearch/xpack/transform/integration/TransformUsageIT.java
+++ b/x-pack/plugin/transform/qa/single-node-tests/src/test/java/org/elasticsearch/xpack/transform/integration/TransformUsageIT.java
@@ -22,6 +22,8 @@ import java.util.concurrent.TimeUnit;
 
 import static org.elasticsearch.xpack.core.transform.TransformField.INDEX_DOC_TYPE;
 import static org.elasticsearch.xpack.transform.TransformInfoTransportAction.PROVIDED_STATS;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
 
 public class TransformUsageIT extends TransformRestTestCase {
 
@@ -102,12 +104,21 @@ public class TransformUsageIT extends TransformRestTestCase {
                     || statName.equals(TransformIndexerStats.SEARCH_TIME_IN_MS.getPreferredName())) {
                     continue;
                 }
-                assertEquals(
-                    "Incorrect stat " + statName,
-                    expectedStats.get(statName).doubleValue(),
-                    extractStatsAsDouble(XContentMapValues.extractValue("transform.stats." + statName, statsMap)),
-                    0.0001
-                );
+
+                // the trigger count can be higher if the scheduler kicked before usage has been called, therefore check for gte
+                if (statName.equals(TransformIndexerStats.NUM_INVOCATIONS.getPreferredName())) {
+                    assertThat(
+                        "Incorrect stat " + statName,
+                        extractStatsAsDouble(XContentMapValues.extractValue("transform.stats." + statName, statsMap)),
+                        greaterThanOrEqualTo(expectedStats.get(statName).doubleValue())
+                    );
+                } else {
+                    assertThat(
+                        "Incorrect stat " + statName,
+                        extractStatsAsDouble(XContentMapValues.extractValue("transform.stats." + statName, statsMap)),
+                        equalTo(expectedStats.get(statName).doubleValue())
+                    );
+                }
             }
             // Refresh the index so that statistics are searchable
             refreshIndex(TransformInternalIndexConstants.LATEST_INDEX_VERSIONED_NAME);

